pkg/runtime/runtime.go

```diff
@@ -758,6 +780,46 @@ func (a *DaprRuntime) initPubSub() error {
 	return nil
 }
 
+// Publish is an adapter method for the runtime to pre-validate publish requests
+// And then forward them to the Pub/Sub component.
+// This method is used by the HTTP and gRPC APIs.
+func (a *DaprRuntime) Publish(req *pubsub.PublishRequest) error {
+	if allowed := a.isPubSubOperationAllowed(req.Topic, a.allowedPublishings); !allowed {
+		return fmt.Errorf("topic %s is not allowed for app id %s", req.Topic, a.runtimeConfig.ID)
+	}
+	return a.pubSub.Publish(req)
+}
+
+func (a *DaprRuntime) isPubSubOperationAllowed(topic string, topicsList []string) bool {
+	allowed := false
+
+	// first check if allowedTopics contain it
+	if len(a.allowedTopics) > 0 {
+		for _, t := range a.allowedTopics {
+			if t == topic {
+				allowed = true
```

Comment:
if its in allowedList, we set it to true, but then below we also check for granular scope.
As per the spec [here](https://github.com/dapr/dapr/issues/1294#issuecomment-601672088) , allowed list overrides granular. Shout it just return true form here?

Reply:
No, allowed list means that if its not in a scope, the scope doesn't apply. if its in allowed, its still subject to specific scopres.

Reply:
got it. 
What would be the behavior in this case?
TopicA is in allowedTopics but its not in publishingScopes. (Currently it would return false)

Reply:
Yeah I just added a test for it and caught that scenario. it should be true

Reply:
Done, please re-check.